src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/StaffCategoryCaseTest.java

```diff
@@ -494,6 +496,167 @@ void shouldDeleteApprovedRequestedRoleForCase_V1_1() {
+                                                             "1234567890123456", DELETE_REQUESTED);
+        requestedRole1.setClassification(Classification.RESTRICTED);
+        requestedRole1.getAttributes().putAll(Map.of("jurisdiction", convertValueJsonNode("IA"),
+                                                     "caseType", convertValueJsonNode("Asylum"),
+                                                     "caseId", convertValueJsonNode("1234567890123456")));
+        assignmentRequest.setRequestedRoles(List.of(requestedRole1));
+        assignmentRequest.getRequest().setClientId(clientId);
+
+        FeatureFlag featureFlag  =  FeatureFlag.builder().flagName(FeatureFlagEnum.CCD_1_0.getValue())
+            .status(true).build();
+        featureFlags.add(featureFlag);
+
+        executeDroolRules(List.of(buildExistingRoleForIAC(requestedRole1.getActorId(),
+                                                          "senior-tribunal-caseworker"),
+                                  buildExistingRoleForIAC(assignmentRequest.getRequest().getAssignerId(),
+                                                          "tribunal-caseworker"),
+                                  buildExistingRoleForIAC(assignmentRequest.getRequest().getAssignerId(),
+                                                          roleName)));
+        //assertion
+        assignmentRequest.getRequestedRoles().forEach(ra -> assertEquals(Status.DELETE_APPROVED, ra.getStatus()));
+    }
+
+    @Test
+    void shouldApproveDefaultDeleteCaseRequestedRoles_CCD_1_0_FromCCD() {
+        verifyDefaultCaseRequestRoleDelete_CCD_1_0(true, Status.DELETE_APPROVED);
+        verifyDefaultCaseRequestRoleDelete_CCD_1_0(false, Status.DELETE_REJECTED);
+    }
+
+    private void verifyDefaultCaseRequestRoleDelete_CCD_1_0(Boolean enableByPass, Status status) {
+        RoleAssignment requestedRole1 = getRequestedCaseRole(RoleCategory.SYSTEM, "[RESPSOLICITOR]",
+                                                             SPECIFIC, "caseId",
+                                                             "1234567890123456", DELETE_REQUESTED);
+        requestedRole1.setClassification(Classification.RESTRICTED);
+        requestedRole1.getAttributes().putAll(Map.of("jurisdiction", convertValueJsonNode("IA"),
+                                                     "caseType", convertValueJsonNode("Asylum"),
+                                                     "caseId", convertValueJsonNode("1234567890123456")));
+        assignmentRequest.setRequestedRoles(List.of(requestedRole1));
+        assignmentRequest.getRequest().setClientId("ccd_data");
+        assignmentRequest.getRequest().setByPassOrgDroolRule(enableByPass);
+
+        FeatureFlag featureFlag  =  FeatureFlag.builder().flagName(FeatureFlagEnum.CCD_1_0.getValue())
+            .status(true).build();
+        featureFlags.add(featureFlag);
+
+        executeDroolRules(List.of(buildExistingRoleForIAC(requestedRole1.getActorId(),
```

Comment:
Use buildExecuteKieSession() instead of executeDroolRule(List< ExistingRoleAssignments>) as we don't need to inject ExistingRoleAssignments to validate CCD Roles in drool. it will unnecessary consume the system memory.